tooling/minimize/src/program.rs

```diff
@@ -64,6 +64,17 @@ impl<'tcx> Ctxt<'tcx> {
             globals: self.globals,
         }
     }
+
+    // Returns FnName associated with some key. If it does not exist it creates a new one.
+    pub fn get_fn_name_or_set(&mut self, key: (rs::DefId, &'tcx rs::List<rs::GenericArg<'tcx>>)) -> FnName {
```

Comment:
I think just `get_fn_name` is a good name. The fact that this might lazily extend the map is an implementation detail.